### ISSUE TITLE
Configurable domains to allow dns-rebinding

### ISSUE THREAD

**Issue by kamermans** [systemd/systemd#15286]
**Is your feature request related to a problem? Please describe.**
As a developer, I use real SSL certificates on my dev projects, so I use public domains pointed at  private, non-routable IPs, like `iot-panel.srkdev.com.`.

Systemd-resolved (at least on an Ubuntu 19.10 Desktop fresh install) does not resolve this address, probably in an effort to mitigate dns-rebinding attacks, which is understandable (although I see no evidence of this mitigation in documentation, config files or the source code).
```
# DNS Record:
# iot-panel.srkdev.com.  IN  A  192.168.86.10

$ nslookup iot-panel.srkdev.com.
Server:		127.0.0.53
Address:	127.0.0.53#53

Non-authoritative answer:
*** Can't find iot-panel.srkdev.com: No answer
```

Here's some debug data in case this is actually a bug, not a feature:
```
Mar 31 12:33:07 steve-laptop-linux systemd-resolved[28457]: Transaction 26648 for <iot-panel.srkdev.com IN AAAA> scope dns on wlp2s0/*.
Mar 31 12:33:07 steve-laptop-linux systemd-resolved[28457]: Using feature level UDP for transaction 26648.
Mar 31 12:33:07 steve-laptop-linux systemd-resolved[28457]: Using DNS server 192.168.86.1 for transaction 26648.
Mar 31 12:33:07 steve-laptop-linux systemd-resolved[28457]: Sending query packet with id 26648.
Mar 31 12:33:07 steve-laptop-linux systemd-resolved[28457]: Processing query...
Mar 31 12:33:07 steve-laptop-linux systemd-resolved[28457]: Processing incoming packet on transaction 26648 (rcode=SUCCESS).
Mar 31 12:33:07 steve-laptop-linux systemd-resolved[28457]: Added NODATA cache entry for iot-panel.srkdev.com IN AAAA 899s
Mar 31 12:33:07 steve-laptop-linux systemd-resolved[28457]: Transaction 26648 for <iot-panel.srkdev.com IN AAAA> on scope dns on wlp2s0/* now complete with <success> from network (unsigned).
Mar 31 12:33:07 steve-laptop-linux systemd-resolved[28457]: Sending response packet with id 37620 on interface 1/AF_INET.
Mar 31 12:33:07 steve-laptop-linux systemd-resolved[28457]: Freeing transaction 26648.
```

**Describe the solution you'd like**
I would like configuration options to be able to:
 - Specify which domains should be allowed to do dns-rebinding
 - Disable this feature entirely

**Describe alternatives you've considered**
After scouring the `systemd-resolved` source code, I was unable to find where this dns-rebind protection is happening.  The only feasible alternative for me (and my entire company) is to disable `systemd-resolved` and use something else like `dnsmasq`.

**Comment by mcginty** [comment]
This seems unrelated to caching as well, setting `Cache = no` doesn't fix it.

**Comment by poettering** [comment]
Judging by the log snippet you posted your DNS server simply response with NODATA on that AAAA lookup?

but you first snippet suggests you care about A lookups? Can you provide a matching log output for that?

**Comment by kamermans** [comment]
Note that I have reinstalled Ubuntu 20.04 on this machine recently and retested this.  It seems to be returning the IPs as expected now:

```
Aug 08 19:56:41 steve-laptop-linux systemd-resolved[68312]: Got message type=method_call sender=:1.475 destination=org.freedesktop.resolve1 path=/org/freedesktop/resolve1 interface=org.freedesktop.resolve1.Manager member=ResolveHostname cookie=2 reply_cookie=0 signature=isit error-name=n/a error-message=n/a
Aug 08 19:56:41 steve-laptop-linux systemd-resolved[68312]: idn2_lookup_u8: iot-panel.srkdev.com → iot-panel.srkdev.com
Aug 08 19:56:41 steve-laptop-linux systemd-resolved[68312]: Looking up RR for iot-panel.srkdev.com IN A.
Aug 08 19:56:41 steve-laptop-linux systemd-resolved[68312]: Looking up RR for iot-panel.srkdev.com IN AAAA.
Aug 08 19:56:41 steve-laptop-linux systemd-resolved[68312]: Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=AddMatch cookie=4 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Aug 08 19:56:41 steve-laptop-linux systemd-resolved[68312]: Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=GetNameOwner cookie=5 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Aug 08 19:56:41 steve-laptop-linux systemd-resolved[68312]: Got message type=method_return sender=org.freedesktop.DBus destination=:1.474 path=n/a interface=n/a member=n/a cookie=7 reply_cookie=5 signature=s error-name=n/a error-message=n/a
Aug 08 19:56:41 steve-laptop-linux systemd-resolved[68312]: Switching to DNS server 192.168.88.22 for interface enp0s31f6.
Aug 08 19:56:41 steve-laptop-linux systemd-resolved[68312]: Cache miss for iot-panel.srkdev.com IN A
Aug 08 19:56:41 steve-laptop-linux systemd-resolved[68312]: Transaction 26824 for <iot-panel.srkdev.com IN A> scope dns on enp0s31f6/*.
Aug 08 19:56:41 steve-laptop-linux systemd-resolved[68312]: Using feature level UDP+EDNS0 for transaction 26824.
Aug 08 19:56:41 steve-laptop-linux systemd-resolved[68312]: Using DNS server 192.168.88.22 for transaction 26824.
Aug 08 19:56:41 steve-laptop-linux systemd-resolved[68312]: Sending query packet with id 26824.
Aug 08 19:56:41 steve-laptop-linux systemd-resolved[68312]: Got message type=method_return sender=org.freedesktop.DBus destination=:1.474 path=n/a interface=n/a member=n/a cookie=6 reply_cookie=4 signature=n/a error-name=n/a error-message=n/a
Aug 08 19:56:41 steve-laptop-linux systemd-resolved[68312]: Match type='signal',sender='org.freedesktop.DBus',path='/org/freedesktop/DBus',interface='org.freedesktop.DBus',member='NameOwnerChanged',arg0=':1.475' successfully installed.
Aug 08 19:56:44 steve-laptop-linux systemd-resolved[68312]: Connection failure for DNS UDP packet: No route to host
Aug 08 19:56:44 steve-laptop-linux systemd-resolved[68312]: Retrying transaction 26824.
Aug 08 19:56:44 steve-laptop-linux systemd-resolved[68312]: Switching to DNS server 192.168.88.1 for interface enp0s31f6.
Aug 08 19:56:44 steve-laptop-linux systemd-resolved[68312]: Cache miss for iot-panel.srkdev.com IN A
Aug 08 19:56:44 steve-laptop-linux systemd-resolved[68312]: Transaction 26824 for <iot-panel.srkdev.com IN A> scope dns on enp0s31f6/*.
Aug 08 19:56:44 steve-laptop-linux systemd-resolved[68312]: Using feature level UDP+EDNS0 for transaction 26824.
Aug 08 19:56:44 steve-laptop-linux systemd-resolved[68312]: Using DNS server 192.168.88.1 for transaction 26824.
Aug 08 19:56:44 steve-laptop-linux systemd-resolved[68312]: Sending query packet with id 26824.
Aug 08 19:56:44 steve-laptop-linux systemd-resolved[68312]: Processing incoming packet on transaction 26824 (rcode=SUCCESS).
Aug 08 19:56:44 steve-laptop-linux systemd-resolved[68312]: Verified we get a response at feature level UDP+EDNS0 from DNS server 192.168.88.1.
Aug 08 19:56:44 steve-laptop-linux systemd-resolved[68312]: Added positive unauthenticated cache entry for iot-panel.srkdev.com IN A 299s on enp0s31f6/INET/192.168.88.1
Aug 08 19:56:44 steve-laptop-linux systemd-resolved[68312]: Transaction 26824 for <iot-panel.srkdev.com IN A> on scope dns on enp0s31f6/* now complete with <success> from network (unsigned).
Aug 08 19:56:44 steve-laptop-linux systemd-resolved[68312]: Sent message type=method_return sender=n/a destination=:1.475 path=n/a interface=n/a member=n/a cookie=6 reply_cookie=2 signature=a(iiay)st error-name=n/a error-message=n/a
Aug 08 19:56:44 steve-laptop-linux systemd-resolved[68312]: Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=RemoveMatch cookie=7 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Aug 08 19:56:44 steve-laptop-linux systemd-resolved[68312]: Freeing transaction 26824.
```

Note that my primary DNS server (`192.168.88.22`) is down, but that was expected.

Here is my version as reported by `resolvectl --version`:
```
systemd 245 (245.4-4ubuntu3.11)
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid
```
